catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -224,20 +187,18 @@ private Object configure() {
   @SuppressWarnings(
       "squid:S1172" /* The code parameter is required in blueprint-cm-1.0.7. See https://issues.apache.org/jira/browse/ARIES-1436. */)
   public void destroy(int code) {
-    if (routeCollection == null) {
+    if (routeBuilder == null) {
       return;
     }
 
-    for (RouteDefinition routeDef : routeCollection) {
+    for (RouteDefinition routeDef : routeBuilder.getRouteCollection().getRoutes()) {
       try {
         String routeId = routeDef.getId();
-        if (isMyRoute(routeId)) {
-          LOGGER.trace("Stopping route with ID = {} and path {}", routeId, monitoredDirectory);
-          camelContext.stopRoute(routeId);
-          boolean status = camelContext.removeRoute(routeId);
-          LOGGER.trace("Status of removing route {} is {}", routeId, status);
-          camelContext.removeRouteDefinition(routeDef);
-        }
+        LOGGER.trace("Stopping route with ID = {} and path {}", routeId, monitoredDirectory);
+        camelContext.stopRoute(routeId);
+        boolean status = camelContext.removeRoute(routeId);
+        LOGGER.trace("Status of removing route {} is {}", routeId, status);
+        camelContext.removeRouteDefinition(routeDef);
```

Comment:
:question: Should `status` be checked before this call?

Reply:
Not necessary, we just use it to log if it was remove or not.